contracts/eosio.bpay/src/eosio.bpay.cpp

```diff
@@ -0,0 +1,73 @@
+#include <eosio.bpay/eosio.bpay.hpp>
+
+namespace eosio {
+
+void bpay::claimrewards( const name owner ) {
+    require_auth( owner );
+
+    rewards_table _rewards( get_self(), get_self().value );
+
+    const auto& row = _rewards.get( owner.value, "no rewards to claim" );
+
+    eosio::token::transfer_action transfer( "eosio.token"_n, { get_self(), "active"_n });
+    transfer.send( get_self(), owner, row.quantity, "producer block pay" );
+
+    _rewards.erase(row);
+}
+
+void bpay::on_transfer( const name from, const name to, const asset quantity, const string memo ) {
+    if (from == get_self() || to != get_self()) {
+        return;
+    }
+
+    // ignore eosio system incoming transfers (caused by bpay income transfers eosio => eosio.bpay => producer)
+    if ( from == "eosio"_n) return;
+
+    symbol system_symbol = eosiosystem::system_contract::get_core_symbol();
+
+    check( quantity.symbol == system_symbol, "only core token allowed" );
+
+    rewards_table _rewards( get_self(), get_self().value );
+    eosiosystem::producers_table _producers( "eosio"_n, "eosio"_n.value );
+
+    eosiosystem::global_state_singleton _global("eosio"_n, "eosio"_n.value);
+    check( _global.exists(), "global state does not exist");
+    uint16_t producer_count = _global.get().last_producer_schedule_size;
```

Comment:
this is handy, makes it more flexible 👍

Reply:
Yeah, I'm not sure what savanna might do to prod count, so...